### ISSUE TITLE
[Feature Request] CHD support

### ISSUE THREAD

**Issue by Sanaki** [jpd002/Play-#837]
Likely not anything even resembling a priority, but support for chd would be a welcome addition. It's quickly becoming a standard for disc-based games in libretro cores for a number of reasons (superior compression ratio, high seek speed, fast compression and extraction, condensation of large filecount cue/cdi/gdi sets, etc), and seems to be gaining support outside of them as well. Speaking personally, the only reason I ever want to use a different format (for discs) is for direct hardware support (primarily PSP with CSO).

Library for directly reading the format: https://github.com/rtissera/libchdr

**Comment by barolo** [comment]
Adding my vote, Dreamcast, PSX, Saturn roms are all chds in my system

**Comment by Foxlum** [comment]
This should actually be a priority, especially with android and iOS support. 4.7 gb's of mostly junk data isn't exactly good on the limited storage.

**Comment by Sanaki** [comment]
While I would obviously adore seeing this implemented, I'm not sure I would call it a high priority at this time. For now, cso should be "good enough" for mobile devices, it's just not -as- good.

**Comment by jpd002** [comment]
So, I read some stuff about the format, and one thing I like about it, is that it combines CUE/BIN files to describe CD-ROM images using a single file. I got a similar complain about dual layer DVDs, there doesn't seem to be a format that can describe them without resorting to having a sidecar file to describe the layout (ie.: the layer break). Do you know if there's any plan to have that inside the format? Can we get in touch with the MAME guys about that?

Another thing: I checked the lib and I don't think it would nicely integrate with the way we handle disc images, so, we would probably end up having to write our own thing inside the emulator (which can be a fun thing to do anyways).

To sum up things, I'm definitively interested to add support for this format inside Play!, but I would be even more interested if the idea of CHD files is to provide some kind of generic disc image format that supports many media types (I saw some stuff about GD-ROMs I there, so I guess that's the intent?). Lemme know if I missed some info, maybe I didn't read things through!

**Comment by Sanaki** [comment]
I'm not -too- familiar with the fine details on the DVD end, but in theory such descriptors may already be possible via custom metadata in the chd (can be done via chdman, though I'm not sure how dangerous this is). The better person to ask about this would undoubtedly be @rtissera, the author of libchdr. Hopefully he can provide some insight. As for asking MAME guys, I'm not really sure how to go about doing that in a way that actually gets a response. They tend to be more on the "PRs welcome" side rather than being interested in specific suggestions.

**Comment by lonkle** [comment]
Really looking forward to this. It seems every disc-based emulator aside from Dolphin is getting CHD support now; can't wait!

**Comment by gouchi** [comment]
Some work have been done[[1](https://github.com/jpd002/Play-/search?q=chd&type=commits)] [[2](https://github.com/jpd002/Play-/blob/master/Source/ui_libretro/main_libretro.cpp#L177)] so tests should be made ;)

**Comment by gouchi** [comment]
And it is working !

@Sanaki If it is ok for you, you may close this issue.

**Comment by Sanaki** [comment]
Sure thing. Thanks for the heads up, I missed that.

**Comment by Sanaki** [comment]
Gave it a quick test, both libretro and standalone. Seems to work well for DVDs, but doesn't work for CDs yet. Tested using Frequency (USA). Not sure if that's known, though the number of CD-based games is fairly low overall.
